modules/core/test/v2/unit/coins/utxo/crossChainRecovery.ts

```diff
@@ -0,0 +1,212 @@
+ *
+ * For instance, a user can receive litecoin on a bitcoin wallet.
+ * This means that the litecoin blockchain has a transaction with outputs that are spendable
+ * with keys that were originally created for a BitGo BTC wallet.
+ * In this example, LTC is the "source coin" and BTC is the "recovery coin"
+ * In cases like these we must use construct a transaction for litecoin network using keys of the
+ * bitcoin wallet.
+ *
+ * @param sourceCoin - the coin to construct the transaction for
+ * @param recoveryCoin - the coin the receiving wallet was set up for
+ */
+function run(sourceCoin: AbstractUtxoCoin, recoveryCoin: AbstractUtxoCoin) {
+  describe(`Cross-Chain Recovery [sourceCoin=${sourceCoin.getChain()} recoveryCoin=${recoveryCoin.getChain()}]`, function () {
+    const recoveryWalletId = '5abacebe28d72fbd07e0b8cbba0ff39e';
+    // the address the accidental deposit went to, in both sourceCoin and addressCoin formats
+    const [depositAddressSourceCoin, depositAddressRecoveryCoin] = [sourceCoin, recoveryCoin].map((coin) =>
+      coin.generateAddress({ keychains: keychainsBase58, index: 0 })
+    );
+    // the address where we want to recover our funds to
+    const recoveryAddress = sourceCoin.generateAddress({ keychains: keychainsBase58, index: 1 }).address;
+    const nocks: nock.Scope[] = [];
+
+    let depositTx: utxolib.bitgo.UtxoTransaction;
+
+    function getDepositUnspents() {
+      return [mockUnspent(sourceCoin.network, 'p2sh', 0, 1e8)];
+    }
+
+    function getDepositTransaction(): utxolib.bitgo.UtxoTransaction {
+      return createFullSignedTransaction(
+        sourceCoin.network,
+        getDepositUnspents(),
+        depositAddressSourceCoin.address,
+        getDefaultWalletUnspentSigner()
+      );
+    }
+
+    before('setup krsProviders', function () {
+      // pretend that Keyternal accepts recoveries for all coins
+      config.krsProviders.keyternal.supportedCoins = ['btc', 'bch', 'bsv', 'bcha', 'ltc', 'zec', 'dash'];
+      config.krsProviders.keyternal.feeAddresses = {
+        tbtc: '2Mujz9eicmgpPcdScRJTywVK3EQNHDJG3yN',
+        tbch: '2Mujz9eicmgpPcdScRJTywVK3EQNHDJG3yN',
+        tbsv: '2Mujz9eicmgpPcdScRJTywVK3EQNHDJG3yN',
+        tbcha: '2Mujz9eicmgpPcdScRJTywVK3EQNHDJG3yN',
+        tltc: 'QffXMViM8DYgPRf1Hoczjw7BS5CVdSWaBL',
+        tzec: 't2ATLAhBP1uTuyiWs5DY5CPH1VuYkGUindt',
+        tdash: '8euHug4dbmPy3CLawwWdeTjGLqPYEGz3Kt',
+      };
+    });
+
+    before('prepare deposit tx', function () {
+      depositTx = getDepositTransaction();
+    });
+
+    before('setup nocks', function () {
+      nocks.push(nockWallet(recoveryCoin, recoveryWalletId, keychainsBase58));
+      nocks.push(nockWalletAddress(recoveryCoin, recoveryWalletId, depositAddressRecoveryCoin));
+      nocks.push(
+        ...nockThirdpartyExplorerTxInfo(sourceCoin, depositTx, getDepositUnspents(), depositAddressSourceCoin.address)
+      );
+    });
+
+    after(function () {
+      nocks.forEach((n) => n.isDone());
```

Comment:
`isDone()` won't throw and will instead return false if the nock isn't consumed, which isn't the behavior you want here I think.

```suggestion
      nocks.forEach((n) => n.done());
```

Reply:
Ah yes of course

Sadly the nocks are persistent and not "done", so the this would throw. I can't find `n.used()` or equivalent.

I'll fix it in a follow-up.